src/app/(auth)/reset-password/page.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import TextInput from '@/components/TextInput/index';
+import { H1, H4 } from '@/styles/text';
+import { OuterDiv, FormDiv, QuestionsDiv } from '@/app/(auth)/styles';
+import BigButton from '@/components/BigButton';
+import supabase from '../../../api/supabase/createClient';
+
+export default function ResetPassword() {
+  const [newPassword, setNewPassword] = useState('');
+  const [newPassword2, setNewPassword2] = useState('');
+  const { push } = useRouter();
```

Comment:
(3) Let's add in a `useEffect()` here that sets the `canReset` useState to true if we consume a `PASSWORD_RECOVERY` event, which can only be triggered by the magic link that is sent via email.

```
useEffect(() => {
   supabase.auth.onAuthStateChange(async (event, session) => {
     if (event == "PASSWORD_RECOVERY") {
       // set the `canReset` useState to true
     }
   })
 }, [])
```